docs/decisions/0003-2FA-for-course-staff-accounts.rst

```diff
@@ -0,0 +1,65 @@
+3. 2FA for Course Team Accounts
+-----------------------------------------
+
+Status
+------
+
+Proposed
+
+Context
+-------
+
+Course team accounts have a great deal of access to and control over sensitive information as well as the ability to run arbitrary Javascript on the LMS. An attacker who compromises an educator account could do a large amount of damage. To protect our learners (and our reputation) we should require 2FA or other enhanced account security for all accounts with these elevated permissions.
+
+An attacker should find it difficult to gain access to the privileged aspects of an educator's account (even if in possession of their password).
+
+
+Proposed Solutions
+------------------
+
+Option 1: Enable Two Factor Authentication (2FA)
+
+Option 2: Ask users for their password before specific restricted actions on site
+
+Option 3: Reset password after a specific time period
+
+
+Decision
+--------
+
+We have decided to go with Option 1. We will enable **Two Factor Authentication** (2FA) for user accounts that belong to course teams (in some cases these can be staff accounts as well).
+
+This change can be accommodated within our current login flow. A high level flow of this process is as follow:
+
+- A user belonging to a course team will authenticate themselves like they currently do, either by providing login credentials or by SSO.
```

Comment:
Is there a definition for the roles that would be appropriate for 2FA, and would that affect the best kind of 2FA for the platform?

As a user/support person, off the top of my head: 
 - course staff, administrator, and course data researcher roles have high levels of access to sensitive data/tools
 - course discussion roles are orthogonal to the above, and give some access to discussion data only
 - beta testers do not have access to learner data, just early access to content
 - a couple roles may exist that are affiliated with an organization rather than a course ("universal course researcher")

Reply:
Currently, we are doing 2FA for course teams and I don't think it would have any effect on the best Kinds of 2FA for the platform, we have just updated the document, please have a look. Thankyou